pkg/segment/structs/segstructs.go

```diff
@@ -961,6 +973,201 @@ func (qtype QueryType) String() string {
 	}
 }
 
+func (qa *QueryAggregators) HasStatsBlock() bool {
+	return qa != nil && !qa.HasStreamStats() && qa.HasGroupByOrMeasureAggsInBlock()
+}
+
+func (qa *QueryAggregators) HasStatsBlockInChain() bool {
+	if qa == nil {
+		return false
+	}
+
+	if qa.HasStatsBlock() {
+		return true
+	}
+
+	if qa.Next != nil {
+		return qa.Next.HasStatsBlockInChain()
+	}
+
+	return false
+}
```

Comment:
nit: there's a helper function `HasInChain()` that will reduce this boilerplate code (see `HasQueryAggergatorBlockInChain()`)